quickwit-metastore/src/metastore/postgresql_metastore.rs

```diff
@@ -51,18 +52,18 @@ mod pg_error_code {
 }
 
 /// Establishes a connection to the given database URI.
-async fn establish_connection(database_uri: &str) -> MetastoreResult<Pool<Postgres>> {
+async fn establish_connection(connection_uri: &Uri) -> MetastoreResult<Pool<Postgres>> {
```

Comment:
Generalizing the use of `Uri` in the metastore implementation so that we don't inadvertently display the credentials when manipulating the URI as a raw string.